src/odemis/driver/test/symphotime_test.py

```diff
@@ -0,0 +1,113 @@
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+
+from __future__ import division
+
+import logging
+from odemis.driver import symphotime
+import unittest
+import os
+import time
+
+logging.getLogger().setLevel(logging.DEBUG)
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)s:%(lineno)d %(message)s")
+
+# Export TEST_NOHW=1 to force using only the simulator and skipping test cases
+# needing real hardware
+TEST_NOHW = (os.environ.get("TEST_NOHW", 0) != 0)  # Default to Hw testing
+
+CONFIG_SCANNER = {
+    "name": "Test scanner",
+    "role": "scanner",
+    "parent": None,
+}
+
+# Config to use with the symphotime simulator
+CONFIG_SYMPHOTIME_SIM = {
+    "name": "Test Symphotime Controller",
+    "role": "detector",
+    "children": { "scanner": CONFIG_SCANNER },
+    "host": "localhost",
+    }
+
+
+class TestSymphotime(unittest.TestCase):
+    """
+    Tests 
+    """
+    @classmethod
+    def setUpClass(self):
+        # note: The symphotime simulator must be running before we start the tests. 
+        self.controller = symphotime.Controller(**CONFIG_SYMPHOTIME_SIM)
+        self.finished = False
+
+    @classmethod
+    def tearDownClass(self):
+        self.controller.terminate()
+    
+    def test_acquisition(self):
+        # Test the starting and stopping of acquisition. 
+        
+        for filename in ["file1.ptu", "big$file.ptu"]:
+            for directory in ["MyMeasurement", "TestGroup"]:
+                self.controller.scanner.filename.value = filename
+                self.controller.scanner.directory.value = directory
+
+                # This function will execute once the measurement completes.
+                def callback(flow, array):
+                    self.assertEqual(array, [[0]])
+                    self.assertEqual(self.controller.scanner.filename.value, filename)
+                    self.assertEqual(self.controller.scanner.directory.value, directory)
+
+                # Start measuring
+                self.controller.data.subscribe(callback)
+
+                # Should start measuring now
+                self.assertTrue(self.controller.isMeasuring())
+
+                # Wait until the measurement completes
+                while self.controller.isMeasuring():
+                    time.sleep(0.1)
+
+                self.controller.data.unsubscribe(callback)
+            
+    def test_stopping(self):
+        # Start measuring, then immediately stop. Be sure that the measurement takes at least 5 seconds.
+        
+        # This function will execute once the measurement completes.
+        def callback(flow, array):
+            raise RuntimeError("This is not supposed to run, since measurement is supposed to stop ")
```

Comment:
Also log, and set a special flag to ensure the test will fail in such case. (because a failure in a callback will not fail the test case normally).